cpp/src/reductions/reduction_functions.cuh

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_REDUCTION_FUNCTIONS_CUH
+#define CUDF_REDUCTION_FUNCTIONS_CUH
+
+#include <cudf/cudf.h>
+#include <cudf/reduction.hpp>
+#include "reduction_operators.cuh"
+
+#include <rmm/rmm.h>
+#include <utilities/cudf_utils.h>
+#include <utilities/error_utils.hpp>
+#include <utilities/type_dispatcher.hpp>
+
+#include <cub/device/device_reduce.cuh>
+
+namespace cudf {
+namespace reductions {
+
+// compute reduction by the operator
+template <typename Op, typename InputIterator, typename T_output>
+void reduction_op(T_output* dev_result, InputIterator d_in, gdf_size_type num_items,
```

Comment:
I feel like this function declaration/definition should go somewhere else.

Reply:
Where? In a separate header?

Reply:
Probably one of the other reduction headers.

Reply:
separated to `reduction.cuh`